File: DaS.ScriptLib/Game/Data/Structures/Map.vb
Imports DaS.ScriptLib.Injection

Namespace Game.Data.Structures

    Public Class Map

        Public Const BasePointer As Integer = &H137D644

        Public Shared ReadOnly Property Pointer As Integer
            Get
                Return RInt32(BasePointer)
            End Get
        End Property

        Public Shared ReadOnly Property CharData1Address As Integer
            Get
                Return RInt32(Pointer + &H3C)
            End Get
        End Property

        Public Shared ReadOnly Property MapEntryCount As Integer
            Get
                Return RInt32(Pointer + &H70)
            End Get
        End Property

        Public Shared Function GetEntries() As MapEntry()
            Dim result(MapEntryCount - 1) As MapEntry
            For i = 0 To result.Length - 1
                result(i) = New MapEntry(Pointer + &H74 + (4 * i))
            Next
            Return result
        End Function

    End Class

End Namespace

File: DaS.ScriptLib.CS/Game/Data/Structures/Map.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Linq;
using System.Xml.Linq;
using System.Threading.Tasks;
using DaS.ScriptLib.Injection;

namespace DaS.ScriptLib.Game.Data.Structures
{

	public class Map
	{


		public const int BasePointer = 0x137d644;
		public static int Pointer {
			get { return Hook.RInt32(BasePointer); }
		}

		public static int CharData1Address {
			get { return Hook.RInt32(Pointer + 0x3c); }
		}

		public static int MapEntryCount {
			get { return Hook.RInt32(Pointer + 0x70); }
		}

		public static MapEntry[] GetEntries()
		{
			MapEntry[] result = new MapEntry[MapEntryCount];
			for (i = 0; i <= result.Length - 1; i++) {
				result[i] = new MapEntry(Pointer + 0x74 + (4 * i));
			}
			return result;
		}

	}

}
